### PR TITLE
Add trend config flow

### PR DESCRIPTION
**Pull request by jpbede**
## Proposed change
Add config flow to `trend` integration. Also add myself as code owner.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/99761
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/trend.markdown [hunk 1]
```diff
@@ -21,6 +24,10 @@ as part of automations, where you want to base an action on a trend.
 
 ## Configuration
```
Comment (gjohansson-ST) [inline source/_integrations/trend.markdown]:
```suggestion
```
the heading is written by the config flow import

**Review by gjohansson-ST — CHANGES_REQUESTED** [review]
Needs rebase

File: source/_integrations/trend.markdown [hunk 2]
```diff
@@ -21,7 +22,9 @@ at least two updates of the underlying sensor to establish a trend.
 Thus it can take some time to show an accurate state. It can be useful
 as part of automations, where you want to base an action on a trend.
 
-## Configuration
+{% include integrations/config_flow.md %}
+
```
Comment (gjohansson-ST) [inline source/_integrations/trend.markdown:37]: Maybe we should add a config basic section here to explain the fields?
```
{% configuration_basic %}
Name:
  description: ...
...
{% endconfiguration_basic %}

File: source/_integrations/trend.markdown [hunk 3]
```diff
@@ -10,7 +10,8 @@ ha_quality_scale: internal
 ha_domain: trend
 ha_platforms:
   - binary_sensor
-ha_integration_type: integration
+ha_config_flow: true
```
Comment (gjohansson-ST) [inline source/_integrations/trend.markdown:14]: Add it to `Helper` in the `ha_category`

**Review by gjohansson-ST — DISMISSED** [review]

File: source/_integrations/trend.markdown (hunk 2 above)
Comment (jpbede) [reply source/_integrations/trend.markdown:37]: Yes, that could be helpful, I think we should also add a description of the config flow itself.

**Review by frenck — APPROVED** [review]
Thanks, @jpbede 👍 

../Frenck

**Pull request merged**